temporal-sdk/src/main/java/io/temporal/client/WorkflowExecutionDescription.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client;
+
+import io.temporal.api.workflowservice.v1.DescribeWorkflowExecutionResponse;
+import io.temporal.common.converter.DataConverter;
+import javax.annotation.Nonnull;
+
+/** Contains information about a workflow execution. */
+public class WorkflowExecutionDescription extends WorkflowExecutionMetadata {
+  private final @Nonnull DescribeWorkflowExecutionResponse response;
+
+  public WorkflowExecutionDescription(
```

Comment:
Can we make this constructor package private like its super class? I assume this is public because it is interceptable? Not sure the describe call needs to be interceptable if the list call isn't. Or if we want to make the list call interceptable, we can and then expose the constructor of `WorkflowExecutionMetadata`.

Reply:
Yes it is public because it is intercept-able, It was a requirement on the issue. The list call is not intercept-able right now but it makes sense to open an issue to add that as well and expose the constructor of WorkflowExecutionMetadata, but I would prefer to do that in a separate issue.

Reply:
Opened an issue to track making list intercept-able https://github.com/temporalio/sdk-java/issues/2328